modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/NodeImpl.java

```diff
@@ -973,7 +973,11 @@ public boolean init(final NodeOptions opts) {
             LOG.error("Node {} initMetaStorage failed.", getNodeId());
             return false;
         }
-        if (!initFSMCaller(new LogId(0, 0))) {
+
+        long lastAppliedIndex = opts.getLastAppliedIndex();
+        long persistedTerm = metaStorage.getTerm();
+
+        if (!initFSMCaller(new LogId(lastAppliedIndex, persistedTerm))) {
```

Comment:
This is dangerous to set term higher than 0 or 1 to a new raft node, this may cause inconsistencies to the raft protocol. I highly recommend to use `org.apache.ignite.raft.jraft.core.NodeImpl#bootstrap`, where guaranties for a new bootstrapped node are checked.

Reply:
I considered that in latest changes. Thank you!